posts/2024-07-26-crates-io-development-update.md

```diff
@@ -0,0 +1,69 @@
+
+
+## Dark mode
+
+If your operating system is set to dark mode, you may have noticed that crates.io now automatically switches to a dark user interface theme. This change was made to make the website more comfortable to use for users who prefer dark themes. If you don't like the dark theme, you can still switch back to the light theme by clicking the color theme icon in the top right corner of the page. By default, the theme will be set based on your operating system's theme settings, but you can also override this setting manually.
+
+Similar to GitHub, we now also support dark/light theme support for images in your `README.md` files:
+
+```html
+<picture>
+  <source media="(prefers-color-scheme: dark)" srcset="https://test.crates.io/logo_dark.svg">
+  <img src="https://test.crates.io/logo.svg" alt="logo" width="200">
+</picture>
+```
+
+
+## RSS feeds
+
+Inspired by our friends at the [Python Package Index](https://warehouse.pypa.io/api-reference/feeds.html), we have introduced a couple of experimental RSS feeds for crates.io:
+
+- <https://static.crates.io/rss/crates.xml>: The latest new crates registered on crates.io (the past 60 minutes, but at least 50 new crates).
+- <https://static.crates.io/rss/updates.xml>: The latest version updates on crates.io (the past 60 minutes, but at least 100 version).
+- e.g. <https://static.crates.io/rss/crates/serde.xml>: The latest version updates of the `serde` crate (the past 24 hours, but at least 10 versions).
+
+This will allow you to keep track of the latest crate releases and updates in your favorite RSS reader. The original GitHub issue requested a feed for all the crates you "follow" on crates.io, but we decided that per-crate feeds would be more useful for now. If you have any feedback on this feature, please let us know!
+
+
+## API token expiry notifications
+
+Our crates.io team member [@hi-rustin](https://github.com/hi-rustin) has been very active in improving our API tokens user experience. If you create an API token with an expiry date, you will now receive a notification email three days before the token expires. This will help you to remember to renew your token before it expires and your scripts stop working.
+
+Following this change, he also implemented a way to create API tokens based on existing tokens, which will make it easier to renew tokens without having to reconfigure all the permissions. The user interface on the "API tokens" settings page now shows a "Regenerate" button, which will allow you to copy/clone existing tokens with the same permissions. Similarly, the token expiry notifications will now also contain a link that directly fills in the permissions of the expiring token, so you can easily create a new token with the same permissions.
```

Comment:
> copy/clone existing tokens

This seems to suggest that it would show the exact same token string as when you originally got when creating the token, but "Regenerate" suggests that it creates a new token with the same permissions and revokes the old token. Which of the two is it?

Reply:
The latter one. It will create a new token.